### PR TITLE
Caching

### PR DESCRIPTION
**Pull request by rtk-rnjn**

### CODE REVIEW

File: src/core/Bot.py [hunk 1]
```diff
@@ -76,6 +76,9 @@ def __init__(self, **kwargs):
         self.lockdown_msg: str = None
         self._BotBase__cogs = commands.core._CaseInsensitiveDict()
 
+        self.message_cache: Dict[int, Any] = {}
+        # TODO: Use LRU caching
```
Comment (rtk-rnjn) [inline src/core/Bot.py:80]: Only thing scare me is, the `self.message_cache` will keep growing and growing....

File: src/core/Bot.py [hunk 2]
```diff
@@ -261,19 +264,32 @@ async def db_latency(self):
...
+                if msg:
+                    self.message_cache[msg.id] = msg
+                    return msg
+
     async def send_message(self, channel_id, content, **kwargs):
         await self.http.send_message(channel_id, content, **kwargs)
 
     async def convey_important_message(
-        self, guild: discord.Guild, text: str, *, view=None, title="⚠️__**IMPORTANT**__⚠️"
+        self, guild: discord.Guild, text: str, *, view=None, title="\N{WARNING SIGN}__**IMPORTANT**__\N{WARNING SIGN}"
```
Comment (rtk-rnjn) [inline src/core/Bot.py:292]: I changed cause my IDE wasn't able to parse that emoji :weary:

File: src/core/Bot.py [hunk 3]
```diff
@@ -261,19 +264,32 @@ async def db_latency(self):
...
         except (discord.HTTPException, discord.NotFound):
             return None
         else:
             return _result
 
+    async def get_or_fetch_message(self, channel: discord.Channel, message_id: int) -> Optional[discord.Message]:
+        # caching cause, due to rate limiting 50/1
+        try:
+            return self.message_cache[message_id]
+            # scripting is always fater than `.get()`
```
Comment (rtk-rnjn) [inline src/core/Bot.py:279]: Another reason is, caching won't ever work if using `.get()`. As it always return None. And hence it force us to recheck. And other `if-else` clause

File: src/core/Bot.py (hunk 2 above)
Comment (deadaf) [reply src/core/Bot.py:292]: Mine too 😂

File: src/core/Bot.py (hunk 1 above)
Comment (deadaf) [reply src/core/Bot.py:80]: Rn, I don't think it'll grow enough to cause any impact on the performance of the application itself, I expect its length to be less than 40k  with the current approach (in around 7 days).

I guess implementing TTL cache can be better than this but I am not sure if we can serialize `discord.Message()`. Maybe we can but I am not really sure.

Comment (rtk-rnjn) [reply src/core/Bot.py:80]: I hope you have enough RAM. TTL, MRU, LRU take lot of RAM

**Pull request merged**